pkgs/top-level/all-packages.nix

```diff
@@ -13940,6 +13940,10 @@ with pkgs;
     zig = buildPackages.zig_0_12;
   };
 
+  # A minimal xar is needed to break an infinite recursion between macfuse-stubs and xar.
+  # It is also needed to reduce the amount of unnecessary stuff in the Darwin bootstrap.
+  xarMinimal = callPackage ../by-name/xa/xar/package.nix { e2fsprogs = null; };
```

Comment:
bzip2 and xz are already in the Darwin bootstrap. My goal is not to pull e2fsprogs and (eventually) macfuse-stubs into the Darwin bootstrap. I don’t want to risk ld64 doing something weird in the future if it doesn’t have bzip2 or xz support in its libxar.

> By the way, the infinite recursion doesn’t currently exist, right? It might be best to avoid putting speculative explanation in here that will only be true when a reverted PR has been adapted and merged again.

Will the comment be updated when that PR is merged again?

Reply:
Once this is merged, the macfuse stuff can be re-reverted with a change to swap it to `xarMinimal` to avoid the infinite recursion. Either the original PR author can do it or one of us can (since it had to be reverted to fix staging-next).